archunit/src/main/java/com/tngtech/archunit/core/importer/ClassFileImportRecord.java

```diff
@@ -90,10 +93,18 @@ void setStaticInitializer(String ownerName, DomainBuilders.JavaStaticInitializer
         staticInitializerBuildersByOwner.put(ownerName, builder);
     }
 
-    void addAnnotations(String ownerName, Set<DomainBuilders.JavaAnnotationBuilder> annotations) {
+    void addClassAnnotations(String ownerName, Set<DomainBuilders.JavaAnnotationBuilder> annotations) {
         this.annotationsByOwner.putAll(ownerName, annotations);
     }
 
+    public void addMemberAnnotations(String declaringClassName, String memberName, String descriptor, Set<DomainBuilders.JavaAnnotationBuilder> annotations) {
+        this.annotationsByOwner.putAll(getMemberKey(declaringClassName, memberName, descriptor), annotations);
```

Comment:
I think you need the `owner`, because this is a global collection through the whole import. If you would not have the owner, then e.g. `Customer.getAddress()` and `Order.getAddress()` would overwrite each other.
As for the descriptor, that is due to method overloads, consider

```
@Secured
accessFoo() {..}

@Unsecured
accessFoo(SecurityDelegate s) {..}
```

If you don't have the descriptor in the key, then such methods would override each other with different annotations. 
You can check, if you adjust `getMemberKey(..)` to only take the member name the tests aren't happy :wink: